src/lvl_script_commands.c

```diff
@@ -5456,6 +5458,54 @@ static void set_game_rule_process(struct ScriptContext* context)
         game.conf.rules.game.allies_share_vision = (TbBool)rulevalue;
         panel_map_update(0, 0, gameadd.map_subtiles_x + 1, gameadd.map_subtiles_y + 1);
         break;
+    case 3: //MapCreatureLimit
+        //this one is a special case because it needs to kill of additional creatures
+        SCRIPTDBG(7, "Changing Game Rule '%s' from %d to %d", rulename, game.conf.rules.game.creatures_count, rulevalue);
+        game.conf.rules.game.creatures_count = rulevalue;
+        struct CreatureControl* cctrl;
+        const struct StructureList* slist = get_list_for_thing_class(TCls_Creature);
```

Comment:
I suppose the logic to kill off some could be split of to it's own function, a check could also be added so it's only called when the value is lower then previous one, as to not needlessly loop trough all

Reply:
done